go/multipooler/connstate/connection_state.go

```diff
@@ -19,38 +19,36 @@ import (
 	"sort"
 	"strings"
 	"sync"
-	"sync/atomic"
 
 	"github.com/multigres/multigres/go/pb/query"
 )
 
 // ConnectionState represents the cumulative state of a connection.
-// This includes all state modifiers like session settings, prepared statements,
-// and portals.
+// This includes all state modifiers like session settings and prepared statements.
 //
 // All methods are thread-safe.
 type ConnectionState struct {
 	// mu protects all mutable fields in this struct.
 	mu sync.Mutex
 
+	// User is the current role set via SET ROLE.
+	// This is NOT used for pool bucket routing - only for RLS enforcement.
```

Comment:
The phrasing of this comment feels inaccurate, since we'd need the same code even if Postgres didn't support RLS just to support ordinary roles correctly

Could maybe say "role enforcement" instead of "RLS enforcement"?